src/main/java/org/veupathdb/service/eda/ds/plugin/abundance/AbundanceScatterplotPlugin.java

```diff
@@ -124,16 +124,16 @@ protected void writeResults(OutputStream out, Map<String, InputStream> dataStrea
           DEFAULT_SINGLE_STREAM_NAME + 
        ", by=" + computeEntityIdColName +")");
       connection.voidEval(getVoidEvalVarMetadataMap(DEFAULT_SINGLE_STREAM_NAME, varMap));
-      connection.voidEval("map <- rbind(map, list('id'=veupathUtils::toColNameOrNull(attributes(abundanceDT)$computedVariableDetails)," +
-                                                 "'plotRef'=rep('xAxisVariable', length(attributes(abundanceDT)$computedVariableDetails$variableId))," +
-                                                 "'dataType'=attributes(abundanceDT)$computedVariableDetails$dataType," +
-                                                 "'dataShape'=attributes(abundanceDT)$computedVariableDetails$dataShape");
+      connection.voidEval("map <- rbind(map, list('id'=veupathUtils::toColNameOrNull(attributes(abundanceDT)$computedVariables[[1]]$computedVariableDetails)," +
```

Comment:
for clarity, im suggesting leaving the schema mostly as is for now, except just make computedVariables not an array/ unnamed list since we only have cases w a single entry for now anyway.